python/cudf/cudf/_lib/pylibcudf/io/types.pyx

```diff
@@ -22,7 +28,39 @@ cdef class TableWithMetadata:
     (e.g. column names)
 
     For details, see :cpp:class:`cudf::io::table_with_metadata`.
+
+    Parameters
+    ----------
+    tbl : Table
+        The input table.
+    column_names : list
+        A list of tuples each containing the name of each column
+        and the names of its child columns (in the same format).
+        e.g.
+        [("id", []), ("name", [("first", []), ("last", [])])]
+
     """
+    def __init__(self, Table tbl, list column_names):
+        self.tbl = tbl
+
+        self.metadata.schema_info = move(self._make_column_info(column_names))
+
+    cdef vector[column_name_info] _make_column_info(self, list column_names):
+        cdef vector[column_name_info] col_name_infos
+        cdef column_name_info info
+
+        col_name_infos.reserve(len(column_names))
+
+        for name, child_names in column_names:
+            if not isinstance(name, str):
+                raise ValueError("Column name must be a string!")
+
+            info.name = move(<string> name.encode())
```

Comment:
There are a lot of `move`s in this file. Are they really all necessary? For instance, this `move` seems like a no-op because this is just a string. The `metadata.schema_info` is a `vector[column_name_info]`, where `column_name_info` doesn't contain any `unique_ptr` or any other object that is movable (let alone not-copyable).

Reply:
You generally only need to move vectors if those vectors contain something that requires a move (like a `unique_ptr`). There are cases where things don't require a move but benefit from the move from a perf perspective, but I don't think we have many of those here.